x-pack/plugins/ingest_manager/server/services/datasource.ts

```diff
@@ -4,23 +4,34 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 import { SavedObjectsClientContract } from 'kibana/server';
+import { AuthenticatedUser } from '../../../security/server';
+import { DeleteDatasourcesResponse } from '../../common';
 import { DATASOURCE_SAVED_OBJECT_TYPE } from '../constants';
 import { NewDatasource, Datasource, ListWithKuery } from '../types';
+import { agentConfigService } from './agent_config';
 
 const SAVED_OBJECT_TYPE = DATASOURCE_SAVED_OBJECT_TYPE;
 
 class DatasourceService {
   public async create(
     soClient: SavedObjectsClientContract,
     datasource: NewDatasource,
-    options?: { id?: string }
+    options?: { id?: string; user?: AuthenticatedUser }
   ): Promise<Datasource> {
     const newSo = await soClient.create<Datasource>(
       SAVED_OBJECT_TYPE,
-      datasource as Datasource,
+      {
+        ...datasource,
+        revision: 1,
+      } as Datasource,
```

Comment:
Kind of incorrect, this is not a Datasource there is no id, for agent I have a different type `AgentSOAttributes` for saved object maybe we can use the same pattern here?

Reply:
updated typings here